packages/reference/src/types.ts

```diff
@@ -55,4 +55,4 @@ export type Action =
       slide?: NavigatorSlideInfo;
     }
   | { type: 'delete'; contentTypeId: string; id: string; entity: EntityType }
-  | { type: 'rendered'; entity: EntityType };
+  | { type: 'rendered'; entity?: EntityType };
```

Comment:
Shouldn't we just add another event for the final overall "rendered"? If we listen to this event from the outside we still the `numberOfLinks+1` actions for `rendered` (if I am not mistaken) while we're only interested in the last one without the `entity`.

Reply:
No, in the hook we listen to `rendered` event, but don't emit it further, instead waiting for all `numberOfLinks` events to be triggered to switch the `rendered` state: https://github.com/contentful/field-editors/pull/300/files#diff-07c0b93de1f5dd0e8d47eba16bd83554R28-R34

Once `rendered = true` **state**, we emit the actual `rendered` **event**: https://github.com/contentful/field-editors/pull/300/files#diff-07c0b93de1f5dd0e8d47eba16bd83554R17-R21

Adding a new event like `finalRendered` would require emitting this even in other places instead of `rendered`, or distinguishing between `rendered` coming from other fields and the multiple reference field.